### PR TITLE
`s_maxage` converts to an int

### LINKED ISSUE #2364: `ResponseCacheControl.s_maxage` raises error when set to an int
`Response().cache_control.s_maxage` errors with `None` is not callable

```
from werkzeug.datastructures import ResponseCacheControl

cc = ResponseCacheControl()
cc.s_maxage = 1

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [13], in <cell line: 1>()
----> 1 cc.s_maxage = 1

File /usr/local/lib/python3.10/site-packages/werkzeug/datastructures.py:1958, in cache_control_property.<locals>.<lambda>(x, v)
   1949 def cache_control_property(key, empty, type):
   1950     """Return a new property object for a cache header. Useful if you
   1951     want to add support for a cache extension in a subclass.
   1952 
   1953     .. versionchanged:: 2.0
   1954         Renamed from ``cache_property``.
   1955     """
   1956     return property(
   1957         lambda x: x._get_cache_value(key, empty, type),
-> 1958         lambda x, v: x._set_cache_value(key, v, type),
   1959         lambda x: x._del_cache_value(key),
   1960         f"accessor for {key!r}",
   1961     )

File /usr/local/lib/python3.10/site-packages/werkzeug/datastructures.py:2039, in _CacheControl._set_cache_value(self, key, value, type)
   2037     self[key] = None
   2038 else:
-> 2039     self[key] = type(value)

TypeError: 'NoneType' object is not callable
```

2.0.3 is OK

Flask use case
```
from flask import Response

response = Response()
response.cache_control.s_maxage = 1
```

Environment:

- Python version: 3.10.4
- Werkzeug version: 2.1.0

### PR DESCRIPTION
**Pull request by Yourun-proger**
- fixes #2364 

Checklist:

- [x] Add tests that demonstrate the correct behavior of the change. Tests should fail without the change.
- [ ] Add or update relevant docs, in the docs folder and in code.
- [x] Add an entry in `CHANGES.rst` summarizing the change and linking to the issue.
- [x] Add `.. versionchanged::` entries in any relevant code docs.
- [x] Run `pre-commit` hooks and fix any issues.
- [x] Run `pytest` and `tox`, no tests failed.

### CODE REVIEW

**Comment by davidism** [comment]
This only addresses the specific attribute, there are others that have the same issue where `type` is set to `None`, and can be addressed at a higher level.

File: src/werkzeug/datastructures.py [hunk 1]
```diff
@@ -2090,6 +2090,9 @@ class ResponseCacheControl(_CacheControl):
     you plan to subclass it and add your own items have a look at the sourcecode
     for that class.
 
+    .. versionchanged:: 2.1.1
+        ``max_age`` works correctly with an argument of type int.
```
Comment (allanlei) [inline src/werkzeug/datastructures.py:2094]: This one should be `s_maxage`?  `max_age` is currently ok

File: tests/test_http.py [hunk 2]
```diff
@@ -119,6 +119,9 @@ def test_cache_control_header(self):
         c.max_age = 3.1
         assert c.max_age == 3
         del c.max_age
+        c.max_age = 3
+        assert c.max_age == 3
+        del c.max_age
```
Comment (allanlei) [inline tests/test_http.py:124]: These test should be `c.s_maxage`.  `max_age` in 2.1.0 is OK

File: src/werkzeug/datastructures.py (hunk 1 above)
Comment (Yourun-proger) [reply src/werkzeug/datastructures.py:2094]: Good catch! I will correct this nonsense.

File: tests/test_http.py (hunk 2 above)
Comment (Yourun-proger) [reply tests/test_http.py:124]: And this, too, when I deal with the rest of the fields. Thanks

**Comment by Yourun-proger** [comment]
Oh. I've made all the necessary changes, but I'm not passing tests at this location:
```py
         assert c.no_cache is None
         assert c.private is None
```
Because now these fields are by default casted to bool. And as you know, `bool(None)` is `False`. 
Here's what should I do? Change these tests?

**Comment by davidism** [comment]
I'll figure out a way to deprecate the old behavior.

**Comment by Yourun-proger** [comment]
You? Does this mean that I should do nothing and just push the changes that are already there, despite the fact that the tests will fail?

**Comment by Yourun-proger** [comment]
I deleted that comment, but still I didn't get it. Do **I** need to do anything with failing tests?

**Comment by davidism** [comment]
No, you're fine.

**Comment by Yourun-proger** [comment]
OK. Thanks a lot! I have been trying to implement this myself for the last couple of hours, but there are too many pitfalls in this case ...
I'll post the changes as soon as possible :3

File: src/werkzeug/datastructures.py (hunk 1 above)
Comment (Yourun-proger) [reply src/werkzeug/datastructures.py]: Done now!

File: tests/test_http.py (hunk 2 above)
Comment (Yourun-proger) [reply tests/test_http.py]: Done now!

**Comment by Yourun-proger** [comment]
Wow! I did not understand why git thinks that I changed so much in the docstrings, I just added info about 2.1.1

**Comment by Yourun-proger** [comment]
@davidism, i try to update changelog, when you do your changes. What i need to do for resolve this?
gitaa

**Pull request merged**